extensions/amp-autocomplete/0.1/amp-autocomplete.js

```diff
@@ -447,23 +535,92 @@ export class AmpAutocomplete extends AMP.BaseElement {
    * @private
    */
   inputHandler_() {
+    let inlinePromise = Promise.resolve();
+
     // If the input is the first entry in the field.
     const firstEntry =
       this.userInput_.length === 0 && this.inputElement_.value.length === 1;
     this.userInput_ = this.inputElement_.value;
-    return this.mutateElement(() => {
-      this.filterDataAndRenderResults_(this.sourceData_, this.userInput_);
-      this.toggleResults_(true);
-      if (this.suggestFirst_) {
-        if (!this.detectBackspace_ || firstEntry) {
-          this.updateActiveItem_(1);
-        }
-        // Reset flag.
-        this.detectBackspace_ = false;
+
+    if (this.inline_) {
+      const match = this.getClosestPriorMatch_(this.trigger_);
+      this.triggered_ = !!match;
+
+      if (!this.triggered_) {
+        return this.mutateElement(() => {
+          this.clearAllItems_();
+        });
+      }
+
+      this.match_ = match;
+      this.userInput_ = this.match_[0].slice(this.trigger_.length);
+
+      if (!this.triggered_ && !this.menuTriggered_) {
```

Comment:
I think this conditional block is unreachable since you're returning on `!this.triggered_` above.